### PR TITLE
 MetaCoq 1.0 for Coq 8.16

### PR DESCRIPTION
**Pull request by mattam82**
ci-skip: coq-metacoq-erasure.1.0+8.16 coq-metacoq-pcuic.1.0+8.16 coq-metacoq-safechecker.1.0+8.16 coq-metacoq-template.1.0+8.16 coq-metacoq-translations.1.0+8.16

### CODE REVIEW

File: extra-dev/packages/coq-metacoq-template/coq-metacoq-template.1.0+8.16/opam [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  ["bash" "./configure.sh"]
+  [make "-j" "%{jobs}%" "template-coq"]
+]
+install: [
+  [make "-C" "template-coq" "install"]
+]
+depends: [
+  "stdlib-shims"
+  "ocaml" {>= "4.07.1"}
+  "coq" { > "8.16~" & < "8.17~" }
```
Comment (palmskog) [inline extra-dev/packages/coq-metacoq-template/coq-metacoq-template.1.0+8.16/opam:30]: I don't think the original bound will work out in CI (it seemingly forbids *all* Coq versions), I suggest the standard:
```suggestion
  "coq" {>= "8.16" & < "8.17~" }
```

File: extra-dev/packages/coq-metacoq-template/coq-metacoq-template.1.0+8.16/opam [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  "coq-equations" { = "1.3+8.16" }
```
Comment (palmskog) [inline extra-dev/packages/coq-metacoq-template/coq-metacoq-template.1.0+8.16/opam:31]: How about being more loose here to account for future updates of Equations:
```suggestion
  "coq-equations" {>= "1.3" }
```

File: extra-dev/packages/coq-metacoq-template/coq-metacoq-template.1.0+8.16/opam (hunk 1 above)
Comment (mattam82) [reply extra-dev/packages/coq-metacoq-template/coq-metacoq-template.1.0+8.16/opam:30]: `> "8.16~"` does not exclude all Coq versions AFAIU, just some prereleases, but fine!

**Comment by mattam82** [comment]
@gares I have a strange behavior where the last package in the skip list is not ignored for some reason (the translations one) https://gitlab.com/coq/opam-coq-archive/-/jobs/2673704389
Looking at the script it's not obvious what is wrong.

**Comment by yforster** [comment]
Sorry for being late about this, but I think we should not remove the Equations version. This would mean that future Equations versions commit to still work with this package here, and from experience in the past this was usually not the case.

**Comment by palmskog** [comment]
@yforster but can't we let the opam bench detect this? Recall that we are continuously testing existing packages with latest versions of Coq and their dependencies, and then we change packages if something errors out.

**Comment by silene** [comment]
> Looking at the script it's not obvious what is wrong.

You (or Github) have put a `\r` at the end of your line, which confuses the script.

**Comment by yforster** [comment]
> @yforster but can't we let the opam bench detect this? Recall that we are continuously testing existing packages with latest versions of Coq and their dependencies, and then we change package definitions if something errors out.

I don't understand the opam bench well enough to answer this myself: If @mattam82 releases an Equations beta in `extra-dev`, does the `opam` bench catch that now the MetaCoq package in `released` is broken?

**Comment by mattam82** [comment]
@silene thanks!

**Comment by yforster** [comment]
My focus is on which questions will turn up in Zulip in the end :) If I have both `released` and `extra-dev` enabled (that's a usual situation for users I think) and do `opam update`, I can get into a state where the MetaCoq package is not installable anymore, right?

**Comment by mattam82** [comment]
This finally works. I had to disable the with-test targets as we did not test them in our CI and they were utterly broken (just the Makefile stuff is broken in the opam setting, the files build fine). Fixed by https://github.com/MetaCoq/metacoq/pull/730

**Comment by mattam82** [comment]
@yforster I think it's fine to loosen the equations bound for now, and get reports that things are not co-installable later. I don't release new equations versions that often, and usually we see breakage already in Coq's CI.

**Comment by mattam82** [comment]
We can put back the `with-test` targets with the next release, so from my side @palmskog this is ready to go.

**Pull request merged**